### PR TITLE
Add support for AAP

### PR DESCRIPTION
**Pull request by cdoan1**
* allow missing azure.yml in vars_files
* create keypairs from environment variable passed in

### CODE REVIEW

File: roles/ocp4-cloud-ipi/tasks/azure-infra-nat-gateway.yml [hunk 1]
```diff
@@ -4,21 +4,94 @@
...
+    resource_group: "{{ azure_resource_group }}"
+    name: "{{ azure_natgateway_pip_name }}"
+  register: natgateway_pip_info
+
+# https://management.azure.com/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/natGateways/test-natgateway?api-version=2021-08-01
+# https://docs.microsoft.com/en-us/rest/api/virtualnetwork/nat-gateways/create-or-update
+#
 - name: Create Azure NAT Gateway
-  shell: "az network nat gateway create --resource-group {{ azure_resource_group }} --name {{ azure_natgateway_name }} --public-ip-addresses {{ azure_natgateway_pip_name }}"
+  azure.azcollection.azure_rm_resource:
```
Comment (andreadecorte) [inline roles/ocp4-cloud-ipi/tasks/azure-infra-nat-gateway.yml:24]: ah nice, so this one is for a generic Azure rest call

File: roles/ocp4-cloud-ipi/tasks/azure-infra-nat-gateway.yml [hunk 2]
```diff
@@ -4,21 +4,94 @@
...
+    api_version: "2021-08-01"
+    resource_group: "{{ azure_resource_group }}"
+    provider: network
+    resource_type: natGateways
+    resource_name: "{{ azure_natgateway_name }}"
+    body:
+      location: "westus3"
```
Comment (andreadecorte) [inline roles/ocp4-cloud-ipi/tasks/azure-infra-nat-gateway.yml]: I think we should not pass the location so we take the one of the RG (or at least should be a parameter)

File: roles/ocp4-cloud-ipi/templates/private_ssh_key.pub.j2 [hunk 3]
```diff
@@ -0,0 +1 @@
+{{ private_ssh_key_pub }}
```
Comment (andreadecorte) [inline roles/ocp4-cloud-ipi/templates/private_ssh_key.pub.j2]: should we just call this public_ssh_key ? I think it is less confusing (same for filename)

Comment (cdoan1) [reply roles/ocp4-cloud-ipi/templates/private_ssh_key.pub.j2]: yes

File: roles/ocp4-cloud-ipi/tasks/azure-infra-nat-gateway.yml (hunk 1 above)
Comment (cdoan1) [reply roles/ocp4-cloud-ipi/tasks/azure-infra-nat-gateway.yml:24]: yes, all apis are reachable through rm_resource

File: roles/ocp4-cloud-ipi/tasks/azure-infra-nat-gateway.yml (hunk 2 above)
Comment (cdoan1) [reply roles/ocp4-cloud-ipi/tasks/azure-infra-nat-gateway.yml]: yes, i'm pulling the location from the pip

File: roles/ocp4-cloud-ipi/templates/private_ssh_key.pub.j2 (hunk 3 above)
Comment (cdoan1) [reply roles/ocp4-cloud-ipi/templates/private_ssh_key.pub.j2]: done

**Review by andreadecorte — APPROVED** [review]

**Pull request merged**